### PR TITLE
Bbg git checkout pull

### PR DESCRIPTION
**Pull request by BruceBGordon**
supports optional "pull" in git-checkout task.

### CODE REVIEW

File: wiki/Jobs.md [hunk 1]
```diff
@@ -85,3 +85,50 @@ First create a project. A project is a collection of jobs. Then click the add ic
...
+    "cron_time": "0 0 * * *",
+    "device_tag": "device_type_1",
+    "notify": "example@yahoo.com",
+    "checkout": {
+        "type": "javascript",
+        "task": "git-checkout",
+        "checkout_url": "git@github.com:yahoo/SomeAndroidProject.git",
+        "branch": "master",
+        "pull": ["origin", "branch-to-pull"]
+    },
```
Comment (thekushpatel) [inline wiki/Jobs.md]: Do we need to specify origin? git sets origin by default when you clone. And you already specify the branch. So you would essentially just be merging 'branch-to-pull' into master.

Comment (BruceBGordon) [inline wiki/Jobs.md]: In some cases we could do less. As you pointed out in some cases the pull reduces to a merge. But the current technique opens up the ability pull from somewhere else as well. Perhaps a user  wants to pull from some open-source project that is external to the current repo.

Comment (thekushpatel) [inline wiki/Jobs.md]: Cool. In that case, two things need to be added.
1. In the tasks readme, the new parameter needs to be documented.
2. the parameter should be validated: must be an array of size 2.

File: wiki/Jobs.md [hunk 2]
```diff
@@ -85,3 +85,50 @@ First create a project. A project is a collection of jobs. Then click the add ic
     ]
 }
 ```
+
+### Example with pull request in checkout.
+
```
Comment (thekushpatel) [inline wiki/Jobs.md]: Also, I think you should get rid of this section. Its really redundant. And the parameter is already documented elsewhere.

**Pull request merged**